File: date/3.14/53.cpp
#include <vector>
using namespace std;
class Solution
{
public:
    int maxSubArray(vector<int> &nums)
    {
        if (nums.size() == 0)
        {
            return nullptr;
        }

        int curVal = -0x3f3f3f3f, maxVal = -0x3f3f3f3f;
        for (auto i = nums.begin(); i < nums.end(); i++)
        {
            curVal += *i;
            if (*i > curVal)
            {
                curVal = *i;
            }
            if (curVal > maxVal)
            {
                maxVal = curVal;
            }
        }

        return maxVal;
    }
};

File: date/3.14/53.py

class Solution:
    def maxSubArray(self, nums) -> int:
        if len(nums)==0:
            return

        curVal=float("-inf")
        maxVal=float("-inf")
        for i in range(len(nums)):
            curVal+=nums[i]
            if nums[i]>curVal:
                curVal=nums[i]
            if curVal>maxVal:
                maxVal=curVal
        return maxVal
